### PR TITLE
wasm-metadce: Keep symbols alive if there is any reference to corresponding GOT.mem or GOT.func import

### PR DESCRIPTION
**Pull request by sbc100**
This prevents the DCE of used-via-address-only symbols in emscripten's `MAIN_MODULE=2`
use case which we are starting to use and recommend a lot more.

Part of the fix for https://github.com/emscripten-core/emscripten/issues/13786

### CODE REVIEW

File: src/wasm.h [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include <string>
 #include <vector>
 
+#include "asmjs/shared-constants.h"
```
Comment (kripken) [inline src/wasm.h:35]: why is this needed? if the metadce cpp change needs it, include it there maybe?

**Review by kripken — COMMENTED** [review]
Change looks good aside from the include question.

File: src/wasm.h (hunk 1 above)
Comment (sbc100) [reply src/wasm.h:35]: Sorry, not sure how that happened!

**Review by kripken — APPROVED** [review]
